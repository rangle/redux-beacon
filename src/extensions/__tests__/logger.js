const { logger } = require('../logger');
const { makeStrConsole } = require('../../utils/str-console');

beforeEach(() => {
  /* eslint-disable no-global-assign */
  console = makeStrConsole();
});

const makeAction = () => ({ type: 'ROUTE_CHANGED', payload: '/' });

/* eslint-disable no-console */
describe('logger()', () => {
  it('logs actions and any resulting analytics event', () => {
    const events = [{ hitType: 'pageview', page: '/' }];
    const action = makeAction();
    logger(events, action);
    const output = console.getOutput();
    expect(output).toMatchSnapshot();
  });

  it('logs actions and any resulting analytics events', () => {
    const events = [
      { hitType: 'pageview', page: '/' },
      { hitType: 'pageview', page: '/home' },
    ];
    const action = makeAction();
    logger(events, action);
    const output = console.getOutput();
    expect(output).toMatchSnapshot();
  });

  it('logs events that are saved offline with a helpful message', () => {
    const events = [{ hitType: 'pageview', page: '/' }];
    const action = makeAction();
    logger(events, action, {}, true);
    const output = console.getOutput();
    expect(output).toMatchSnapshot();
  });

  it('logs events that were saved offline with a helpful message', () => {
    const events = [{ hitType: 'pageview', page: '/' }];
    const action = makeAction();
    logger(events, action, {}, false, true);
    const output = console.getOutput();
    expect(output).toMatchSnapshot();
  });

  it('logs nothing when there are no events', () => {
    const events = [];
    const action = makeAction();
    logger(events, action);
    const output = console.getOutput();
    expect(output).toMatchSnapshot();
  });
});
