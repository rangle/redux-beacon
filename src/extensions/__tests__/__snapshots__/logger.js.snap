exports[`logger() logs actions and any resulting analytics event 1`] = `
"%c Analytics
  %c Action:  color: #93748A; font-weight: bold; {\"type\":\"ROUTE_CHANGED\",\"payload\":\"/\"}
  %c Event:  color: #93748A; font-weight: bold; {\"hitType\":\"pageview\",\"page\":\"/\"}"
`;

exports[`logger() logs actions and any resulting analytics events 1`] = `
"%c Analytics
  %c Action:  color: #93748A; font-weight: bold; {\"type\":\"ROUTE_CHANGED\",\"payload\":\"/\"}
%c Events:
  %c (1) color: #93748A; font-weight: bold; {\"hitType\":\"pageview\",\"page\":\"/\"}
  %c (2) color: #93748A; font-weight: bold; {\"hitType\":\"pageview\",\"page\":\"/home\"}"
`;

exports[`logger() logs events that are saved offline with a helpful message 1`] = `
"%c Analytics (pushed to offline storage)
  %c Action:  color: #93748A; font-weight: bold; {\"type\":\"ROUTE_CHANGED\",\"payload\":\"/\"}
  %c Event:  color: #93748A; font-weight: bold; {\"hitType\":\"pageview\",\"page\":\"/\"}"
`;

exports[`logger() logs events that were saved offline with a helpful message 1`] = `
"%c Analytics (from offline storage)
  %c Event:  color: #93748A; font-weight: bold; {\"hitType\":\"pageview\",\"page\":\"/\"}"
`;

exports[`logger() logs nothing when there are no events 1`] = `""`;
